### PR TITLE
Use HtPP for RStrConstPool

### PR DESCRIPTION
**Pull request by thestr4ng3r**
Test: https://github.com/radareorg/radare2-regressions/pull/2010

### CODE REVIEW

**Comment by thestr4ng3r** [comment]
@ret2libc Is it legal to return the keys as the pooled strings here, i.e. assume the keys will stay allocated and keep their contents as long as the HtPP is alive?

File: libr/util/str_constpool.c [hunk 1]
```diff
@@ -2,19 +2,31 @@
...
+	ht_pp_free (pool->ht);
 }
 
 R_API const char *r_str_constpool_get(RStrConstPool *pool, const char *str) {
 	if (!str) {
 		return NULL;
 	}
-	sdb_ht_insert (pool->ht, str, str);
-	return sdb_ht_find (pool->ht, str, NULL);
+	HtPPKv *kv = ht_pp_find_kv (pool->ht, str, NULL);
```
Comment (ret2libc) [inline libr/util/str_constpool.c]: this looks over complicated. What about:

```C
const char *res = ht_pp_find (pool->ht, str, NULL);
if (!res) {
  ht_pp_insert (pool->ht, str, str);
  res = str;
}
return res;
```

**Comment by ret2libc** [comment]
> @ret2libc Is it legal to return the keys as the pooled strings here, i.e. assume the keys will stay allocated and keep their contents as long as the HtPP is alive?

Mh.. Yeah, it should be an ok assumption.

File: libr/util/str_constpool.c (hunk 1 above)
Comment (thestr4ng3r) [reply libr/util/str_constpool.c]: Here res would be the value, not the key. The idea is that I don't have to allocate both the key and the value, but just set the value to NULL and return the key. So I essentially use HtPP as a set, not a map.

Comment (ret2libc) [reply libr/util/str_constpool.c]: Mmm I see why you are doing it. You are doing it because you want to use the key, which is strdupped in HtPP. Oki.

**Pull request merged**